eZ/Publish/API/Repository/Tests/SearchServiceLocationTest.php

```diff
@@ -157,6 +186,35 @@ public function testFieldCollectionContainsNoMatch()
         $this->assertEquals(0, $result->totalCount);
     }
 
+    /**
+     * Test for the findLocations() method.
+     *
+     * @see \eZ\Publish\API\Repository\SearchService::findLocations()
+     */
+    public function testNonPrintableUtf8Characters()
+    {
+        $folder = $this->createFolderWithNonPrintableUtf8Characters();
+        $query = new LocationQuery(
+            [
+                'query' => new Criterion\Field(
+                    'name',
+                    Criterion\Operator::EQ,
+                    utf8_decode("Non\x09Printable\x0EFolder")
```

Comment:
shouldn't we also test for pattern after replacement? If I get the PR right, it should be found as well.

Reply:
It will fail for the legacy search engine, as replacement is not handled there.

Reply:
> It will fail for the legacy search engine, as replacement is not handled there.

Looks like a bug then.

Reply:
Oh, I was thinking is on purpose - let me check it then.

Reply:
@alongosz Legacy SE does not seem to use Field value mappers at all so that seems to be a missing feature which should rather be reported to backlog or something.

Ok if @mateuszbieniek skip this test if not on Solr for no to unblock this? _(this is escalated for solr usage as indexing is broken for these cases)_

Reply:
> @alongosz Legacy SE does not seem to use Field value mappers at all so that seems to be a missing feature which should rather be reported to backlog or something.
> 
> Ok if @mateuszbieniek skip this test if not on Solr for no to unblock this? _(this is escalated for solr usage as indexing is broken for these cases)_

Hmm, ok. It can be skipped for non-Solr, but I'd like to see that case anyway ;)

Reply:
> Hmm, ok. It can be skipped for non-Solr, but I'd like to see that case anyway ;)

Here you go :)